src/client/inputhandler.h

```diff
@@ -285,6 +288,35 @@ class RealInputHandler : public InputHandler
 	{
 		return m_receiver->WasKeyReleased(keycache.key[k]) || joystick.wasKeyReleased(k);
 	}
+	virtual float getMovementSpeed()
+	{
+		if (m_receiver->IsKeyDown(keycache.key[KeyType::FORWARD]) ||
+				m_receiver->IsKeyDown(keycache.key[KeyType::BACKWARD]) ||
+				m_receiver->IsKeyDown(keycache.key[KeyType::RIGHT]) ||
+				m_receiver->IsKeyDown(keycache.key[KeyType::LEFT]))
+			return 1.0; /* If there is a keyboard event, assume maximum speed */
+		return joystick.getMovementSpeed();
+	}
+	virtual float getMovementDirection()
+	{
+		int x = 0;
+		int z = 0;
```

Comment:
float because `atan2` doesn't take ints anyway and to match the return type
```suggestion
		float x = 0, z = 0;
```